Modix.Services/GuildStats/GuildStatSetup.cs

```diff
@@ -0,0 +1,15 @@
+﻿using MediatR;
+using Microsoft.Extensions.DependencyInjection;
+using Modix.Services.Messages.Discord;
+
+namespace Modix.Services.GuildStats
+{
+    public static class GuildStatSetup
+    {
+        public static IServiceCollection AddGuildStats(this IServiceCollection services)
+            => services
+                .AddScoped<IGuildStatService, GuildStatService>()
+                .AddScoped<INotificationHandler<UserJoined>, GuildStatService>()
+                .AddScoped<INotificationHandler<UserLeft>, GuildStatService>();
```

Comment:
Will this create a instance and therefore new cache every time a request is made? And will there be separate instances depending on if the service was resolved as `IGuildStatService`, `INotificationHandler<UserLeft>`, or `INotificationHandler<UserLeft>`?

Reply:
It will create a new instance per-scope, but that's fine as the cache is using IMemoryCache which is singleton scoped, so it'll be persistent.